src/utils/codeBuilder.ts

```diff
@@ -0,0 +1,307 @@
+    static readonly TEMPLATE_VARIABLES = 'TEMPLATE_VARIABLES';
+    static readonly TEMPLATE_VARIABLE_ASSIGNMENTS =
+        'TEMPLATE_VARIABLE_ASSIGNMENTS';
+    static readonly TEMPLATE_VIEW_LWC_LABEL = 'TEMPLATE_VIEW_LWC_LABEL';
+    static readonly TEMPLATE_LWC_NAME = 'TEMPLATE_LWC_NAME';
+    static readonly TEMPLATE_QUICK_ACTION_ICON = 'TEMPLATE_QUICK_ACTION_ICON';
+    static readonly TEMPLATE_QUICK_ACTION_LABEL = 'TEMPLATE_QUICK_ACTION_LABEL';
+
+    private extensionUri: Uri;
+    private objectApiName: string;
+    templateVariables: TemplateVariables;
+    fieldNames: string[];
+
+    constructor(
+        extensionUri: Uri,
+        objectApiName: string,
+        compactLayoutFields: CompactLayoutField[]
+    ) {
+        this.extensionUri = extensionUri;
+        this.objectApiName = objectApiName;
+
+        this.fieldNames = this.getFieldNames(compactLayoutFields);
+        this.templateVariables = this.generateTemplateVariables(
+            this.fieldNames
+        );
+    }
+
+    async generateView(): Promise<boolean> {
+        return new Promise(async (resolve) => {
+            const lwcName = `view${this.objectApiName}Record`;
+            this.copyTemplateFiles(
+                this.templateVariables,
+                'viewRecord',
+                lwcName
+            );
+            this.createQuickAction(this.templateVariables, 'View', lwcName);
+            resolve(true);
+        });
+    }
+
+    async generateEdit(): Promise<boolean> {
+        return new Promise(async (resolve) => {
+            const lwcName = `edit${this.objectApiName}Record`;
+            this.copyTemplateFiles(
+                this.templateVariables,
+                'editRecord',
+                lwcName
+            );
+            this.createQuickAction(
+                this.templateVariables,
+                'Edit',
+                lwcName,
+                'editActionIcon'
+            );
+            resolve(true);
+        });
+    }
+
+    async generateCreate(): Promise<boolean> {
+        return new Promise(async (resolve) => {
+            const lwcName = `create${this.objectApiName}Record`;
+            this.copyTemplateFiles(
+                this.templateVariables,
+                'createRecord',
+                lwcName
+            );
+            this.createQuickAction(this.templateVariables, 'Create', lwcName);
+            resolve(true);
+        });
+    }
+
+    private getFieldNames(compactLayoutFields: CompactLayoutField[]) {
+        const fieldNames: string[] = [];
+        compactLayoutFields.forEach((field) => {
+            field.layoutComponents.forEach((component) => {
+                fieldNames.push(component.value);
+            });
+        });
+        return fieldNames;
+    }
+
+    private createQuickAction(
+        templateVariables: TemplateVariables,
+        label: string,
+        name: string,
+        iconName?: string | undefined
```

Comment:
Remaining nit: you don't need the `| undefined`. The `?:` implies that it's either the specified type, or not provided—which equates to `undefined`.

Reply:
Ahhhhh, sorry! Will update and push!